lib/Controller/SignalingController.php

```diff
@@ -337,11 +356,13 @@ private function backendAuth($auth) {
 		return new JSONResponse($response);
 	}
 
-	private function backendRoom($room) {
-		$roomId = $room['roomid'];
-		$userId = $room['userid'];
+	private function backendRoom($roomRequest) {
+		$roomId = $roomRequest['roomid'];
+		$userId = $roomRequest['userid'];
+		$sessionId = $roomRequest['sessionid'];
+
 		try {
-			$room = $this->manager->getRoomForParticipantByToken($roomId, $userId);
+			$room = $this->manager->getRoomByToken($roomId);
 		} catch (RoomNotFoundException $e) {
 			return new JSONResponse([
```

Comment:
Since this is a OCSController now, you should return `DataResponse` only.
In your client you can set the accept header or append `format=json` to get json:
https://docs.nextcloud.com/server/12/developer_manual/core/externalapi.html#output

Reply:
Changed